airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/JobPersistence.java

```diff
@@ -150,4 +152,39 @@
 
   Optional<Job> getNextJob() throws IOException;
 
+  /// ARCHIVE
+
+  /**
+   * List all available table names in the Database's @param schema
+   */
+  List<String> listTables(String schema) throws IOException;
+
+  /**
+   * Dump the @param tableSQL into a stream of records deserialized as JsonNode objects
+   */
+  Stream<JsonNode> dump(String tableSQL) throws IOException;
+
+  /**
+   * Create a table @param tableName in the @param schema using the @param jsonSchema
+   */
+  void createTable(String schema, String tableName, JsonNode jsonSchema) throws IOException;
+
+  /**
+   * Insert stream of records @param recordStream, described by @param jsonSchema, into the
+   * table @param tableName in the @param schema
+   */
+  void insertRecords(final String tableSchema, final String tableName, final JsonNode jsonSchema, final Stream<JsonNode> recordStream)
+      throws IOException;
+
```

Comment:
Actually I could combine `createTable` and `insertRecords` into one method together...